### PR TITLE
removes the emptydrop results from Optimus in sn_rna mode

### PR DESCRIPTION
**Pull request by kishorikonwar**
### Purpose
We remove the emptydrops results for Optimus in the sn_rna  mode,  but continue to retain in the sc_rna mode.

### Changes
Updated to the new docker container, with the sctools changes,  in the CalcualteCellMetrics task

### CODE REVIEW

File: docker/zarr-output/create_zarr_optimus.py [hunk 1]
```diff
@@ -160,46 +160,65 @@ def add_cell_metrics(
 
     # Read the csv input files
     metrics_df = pd.read_csv(metrics_file, dtype=str)
-    emptydrops_df = pd.read_csv(emptydrops_file, dtype=str)
+ 
+    if add_emptydrops_results == 'yes':
```
[the remark this answers is not in the document]
Comment (kishorikonwar) [reply docker/zarr-output/create_zarr_optimus.py:164]: We could but these variables are then arguments to the python code needs the "yes/no" strings

File: docker/zarr-output/create_zarr_optimus.py [hunk 2]
```diff
@@ -160,46 +160,65 @@ def add_cell_metrics(
...
 
     # Check that input is valid
     if metrics_df.shape[0] == 0 or metrics_df.shape[1] == 0:
         logging.error("Cell metrics table is not valid")
         raise ValueError()
-    if emptydrops_df.shape[0] == 0 or emptydrops_df.shape[1] == 0:
-        logging.error("EmptyDrops table is not valid")
-        raise ValueError()
+
+    if add_emptydrops_results == 'yes':
```
[the remark this answers is not in the document]
Comment (jessicaway) [reply docker/zarr-output/create_zarr_optimus.py:172]: This comment still applies

**Comment by kishorikonwar** [comment]
retest

File: docker/zarr-output/create_zarr_optimus.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kishorikonwar) [reply docker/zarr-output/create_zarr_optimus.py:172]: Yes, I addressed this comment. Sadly, we will be removing, in a subsequent ticket,  this  code as a part of the zarr removal process.

[the remark this answers is not in the document]
Comment (jessicaway) [reply docker/zarr-output/create_zarr_optimus.py:172]: Should this PR be replaced by one that removes empty drops from the corresponding loom file then?

File: docker/zarr-output/create_zarr_optimus.py [hunk 3]
```diff
@@ -160,49 +160,34 @@ def add_cell_metrics(
...
+        if emptydrops_df.shape[0] == 0 or emptydrops_df.shape[1] == 0:
+            logging.error("EmptyDrops table is not valid")
+            raise ValueError()
 
     # Rename cell columns for both datasets to cell_id
-    emptydrops_df = emptydrops_df.rename(columns={"CellId": "cell_id"})
+    if add_emptydrops_results == 'yes':
+       emptydrops_df = emptydrops_df.rename(columns={"CellId": "cell_id"})
```
Comment (jessicaway) [inline docker/zarr-output/create_zarr_optimus.py:179]: It appears that the definition of emptydrops_df is after the validation on line 173

File: docker/zarr-output/create_zarr_optimus.py [hunk 4]
```diff
@@ -255,20 +273,22 @@ def add_cell_metrics(
...
-        else:
-            final_df_bool[index, 0] = np.nan
-
-        if row["emptydrops_IsCell"] == "TRUE":
-            final_df_bool[index, 1] = True
-        elif row["emptydrops_IsCell"] == "FALSE":
-            final_df_bool[index, 1] = False
-        else:
-            final_df_bool[index, 1] = np.nan
+    #if add_emptydrops_results == 'yes':
```
Comment (jessicaway) [inline docker/zarr-output/create_zarr_optimus.py:276]: do we need this condition? if not, please clean up

**Comment by jessicaway** [comment]
I believe this was moved to a separate PR. @kishorikonwar can we close this PR?

**Pull request closed without merging**